### PR TITLE
61 implement switchnetwork when calling the bridge or claim

### LINKED ISSUE #61: Implement switchNetwork when calling the bridge or claim
Implement switchNetwork when calling the bridge or claim in the `BridgeProvider`.

### PR DESCRIPTION
**Pull request by adrifdez**
Closes #61 

### What does this PR does?

Check the network connected and launch the call if it's the correct network. If it's not the proper network is going to open a Metamask window asking to switch to the correct network and if that network isn't setup, it'll offer to add it to your Metamask

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Respect code style and lint
- [ ] Update documentation (if needed)

### CODE REVIEW

File: src/contexts/providers.context.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ const ProvidersProvider: FC = (props) => {
 
           try {
             if (window.ethereum && window.ethereum.isMetaMask) {
-              const web3Provider = new Web3Provider(window.ethereum);
+              const web3Provider = new Web3Provider(window.ethereum, "any");
```
[the remark this answers is not in the document]
Comment (adrifdez) [reply src/contexts/providers.context.tsx:63]: It's related to this issue [https://github.com/ethers-io/ethers.js/issues/866](https://github.com/ethers-io/ethers.js/issues/866)

[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/contexts/providers.context.tsx:63]: https://github.com/ethers-io/ethers.js/issues/866

File: src/views/transaction-confirmation/transaction-confirmation.view.tsx [hunk 2]
```diff
@@ -15,40 +15,55 @@ import Icon from "src/views/shared/icon/icon.view";
...
   const classes = useConfirmationStyles();
-  const [isNetworkIncorrect, setIsNetworkIncorrect] = useState(false);
+  const navigate = useNavigate();
+  const { openSnackbar } = useUIContext();
   const { bridge } = useBridgeContext();
+  const { transaction, setTransaction } = useTransactionContext();
   const { account, connectedProvider } = useProvidersContext();
-  const navigate = useNavigate();
-  const { transaction } = useTransactionContext();
+  const [incorrectMessageNetwork, setIncorrectMessageNetwork] = useState<string>();
```
Comment (amonsosanz) [inline src/views/transaction-confirmation/transaction-confirmation.view.tsx:29]: `[incorrectNetworkMessage, setIncorrectNetworkMessage]`

File: src/views/transaction-details/transaction-details.view.tsx [hunk 3]
```diff
@@ -9,20 +9,25 @@ import { ReactComponent as SpinnerIcon } from "src/assets/icons/spinner.svg";
...
+import routes from "src/routes";
 
 const TransactionDetails: FC = () => {
   const { transactionId } = useParams();
+  const navigate = useNavigate();
   const { openSnackbar } = useUIContext();
   const { getTransactions, claim } = useBridgeContext();
-  const { account } = useProvidersContext();
+  const { account, connectedProvider } = useProvidersContext();
+  const [incorrectMessageNetwork, setIncorrectMessageNetwork] = useState<string>();
```
Comment (amonsosanz) [inline src/views/transaction-details/transaction-details.view.tsx:30]: `[incorrectNetworkMessage, setIncorrectNetworkMessage]`

**Review by elias-garcia — APPROVED** [review]

**Review by amonsosanz — APPROVED** [review]

**Pull request merged**